go/libraries/doltcore/rowconv/super_schema.go

```diff
@@ -0,0 +1,267 @@
+// Copyright 2019 Liquidata, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package rowconv
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/liquidata-inc/dolt/go/libraries/doltcore/doltdb"
+	"github.com/liquidata-inc/dolt/go/libraries/doltcore/schema"
+	"github.com/liquidata-inc/dolt/go/store/hash"
+	"github.com/liquidata-inc/dolt/go/store/types"
+)
+
+// RowConvForSuperSchema creates a RowConverter for transforming rows with the the given schema to the given super schema.
+// This is done by mapping the column tag and type to the super schema column representing that tag and type.
+func RowConvForSuperSchema(sch, super schema.Schema) (*RowConverter, error) {
+	inNameToOutName := make(map[string]string)
+	allCols := sch.GetAllCols()
+	err := allCols.Iter(func(tag uint64, col schema.Column) (stop bool, err error) {
+		inNameToOutName[col.Name] = fmt.Sprintf("%d_%s", col.Tag, col.Kind.String())
+		return false, nil
+	})
+
+	if err != nil {
+		return nil, err
+	}
+
+	fm, err := NewFieldMappingFromNameMap(sch, super, inNameToOutName)
+
+	if err != nil {
+		return nil, err
+	}
+
+	return NewRowConverter(fm)
+}
+
+// TagKindPair is a simple tuple that holds a tag and a NomsKind of a column
+type TagKindPair struct {
+	// Tag is the tag of a column
+	Tag uint64
+
+	// Kind is the NomsKind of a colum
+	Kind types.NomsKind
+}
+
+// NameKindPair is a simple tuple that holds the name of a column and it's NomsKind
+type NameKindPair struct {
+	// Name is the name of the column
+	Name string
+
+	// Kind is the NomsKind of the column
+	Kind types.NomsKind
+}
+
+// SuperSchemaGen is a utility class used to generate the superset of several schemas.
+type SuperSchemaGen struct {
+	tagKindToDestTag map[TagKindPair]uint64
+	usedTags         map[uint64]struct{}
```

Comment:
For existence-check maps like this, I find it's way easier to use bool as the value type. Then you can use the simpler key lookup syntax (instead of _, ok)